localstack/services/s3/s3_starter.py

```diff
@@ -119,7 +119,7 @@ def delete_bucket(self, bucket_name, *args, **kwargs):
         try:
             return delete_bucket_orig(bucket_name, *args, **kwargs)
         except s3_exceptions.MissingBucket:
-            pass
+            raise s3_exceptions.MissingBucket
```

Comment:
I guess we can simply remove the `try`/`except` block here.

Also, seems like a couple of tests are failing past this change - we could create a small test util function with client code to delete a bucket with a `try`/`except` (to ignore the error in case it does not exist).